src/systems.rs

```diff
@@ -128,8 +128,38 @@ pub fn process_input(
     // When a user releases a mouse button, Safari emits both `CursorLeft` and `CursorEntered`
     // events during the same frame. We don't want to reset mouse position in such a case, otherwise
     // we won't be able to process the mouse button event.
-    if cursor_left_window.is_some() && cursor_left_window != cursor_entered_window {
-        egui_context.mouse_position = None;
+    if let Some(window_id) = &cursor_left_window {
+        if cursor_left_window != cursor_entered_window {
```

Comment:
Hi! Thanks for the PR.

Is the goal here to fix the behaviour for any platform? I believe this branch will be entered only for Safari, as this is the only browser that emits `CursorLeft` and `CursorEntered` on mouse button release.

Reply:
That seems off since this change fixed unexpected behavior on desktop linux.

Reply:
Furthermore, this branch needs to be possible to entered on all platforms - after all it's the only branch where `mouse_position` is reset.

Reply:
Hm.. I'm not sure it's working as intended. If I keep dragging the mouse pointer and it leaves the window, it'll fire the release button event immediately. Yet in the `main` version we keep reading mouse events when outside of a window, which allows us to continue dragging sliders for example (see #61).

https://user-images.githubusercontent.com/2943388/190848814-f1d668ab-62d0-4d2b-a88c-514d307b2a3a.mov

Reply:
I'm also not sure I understand what issue the PR is trying to address (or whether the issue indeed exists).

During this recording (the `main` branch), I dragged the cursor outside of the window, released it when being outside, moved the cursor back to the window, and we can see that `bevy_egui` didn't continue drawing to the canvas. If I understand correctly, your point is that we don't fire mouse release events when releasing the cursor outside of the window. But if that was the case, wouldn't it keep drawing when I moved the cursor back to the window?

https://user-images.githubusercontent.com/2943388/190848987-c19e275d-8942-410c-9aa2-4aa1ec5595a0.mov

Reply:
Yes, my main issue is that when the mouse is released outside the window, the release event is never fired, and in a closed-source project i work on it does cause the mean that drawing continues.

Reply:
And I just tested the main branch - the drawing continues when i release the button outside the window and move the cursor back in.

Reply:
[egui-show-bug-.webm](https://user-images.githubusercontent.com/6781733/190850904-b2e274dc-7744-4ba4-ba4d-dbee04e16050.webm)

Reply:
Oh, I see, thanks for clarifying. I was indeed able to reproduce it on my KDE laptop.

Turns out, for some platforms, `winit` fires `CursorLeft` together with button input events when releasing a button outside of a window. I've just pushed  #123 with the fix that also preserves the dragging behaviour. Could you please test it and let me know if it works for you?

Reply:
It seems to work :)

Reply:
Merged and released. Thanks again for flagging it!